Gems/ROS2/Code/Source/Manipulator/JointMotorControllerComponent.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <AzFramework/Entity/EntityDebugDisplayBus.h>
+#include <HingeJointComponent.h>
+#include <PhysX/Joint/PhysXJointRequestsBus.h>
+#include <PrismaticJointComponent.h>
+#include <ROS2/Manipulator/JointMotorControllerComponent.h>
+#include <imgui/imgui.h>
+
+namespace ROS2
+{
+    void JointMotorControllerComponent::Activate()
+    {
+        AZ::TickBus::Handler::BusConnect();
+        ImGui::ImGuiUpdateListenerBus::Handler::BusConnect();
+        AZ::EntityBus::Handler::BusConnect(GetEntityId());
+    }
+
+    void JointMotorControllerComponent::Deactivate()
+    {
+        ImGui::ImGuiUpdateListenerBus::Handler::BusDisconnect();
+        AZ::TickBus::Handler::BusDisconnect();
+    }
+
+    void JointMotorControllerComponent::Reflect(AZ::ReflectContext* context)
+    {
+        JointMotorControllerConfiguration::Reflect(context);
+
+        if (auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<JointMotorControllerComponent, AZ::Component>()->Version(2)->Field(
+                "JointMotorControllerConfiguration", &JointMotorControllerComponent::m_jointMotorControllerConfiguration);
+            if (AZ::EditContext* ec = serializeContext->GetEditContext())
+            {
+                ec->Class<JointMotorControllerComponent>(
+                      "Joint Motor Controller Component", "Base component for motor controller components.")
```

Comment:
The joint motor controller component is not supposed to be displayed as an addable component since it is only a base component class from which "addable" motor controllers derive. When it comes to other, usable controllers, they are already assigned a category.